sympy/integrals/transforms.py

```diff
@@ -1610,7 +1651,7 @@ def _laplace_rule_heaviside(f, t, s, doit=True, **hints):
     This internal helper function tries to transform a product containing the
     `Heaviside` function and returns `None` if it cannot do it.
     """
-    hints.pop('simplify', True)
+    hints.get('simplify', True)
```

Comment:
```suggestion
```

Reply:
I changed this because my introduction of `_use_rules` changes the way recursion works. It will actually improve the effectiveness of using rules recursively. From this point of view, do you still think that the hint simplify should be removed here?

Reply:
I don't think that it affects the recursion, but I think that the problem is that the variable is unused.

Reply:
Ah, true, so I will just omit the line, thanks.